### PR TITLE
scons -> meson

### PR DESCRIPTION
**Pull request by yeled**
This applies to `--HEAD` only for now.

### CODE REVIEW

File: Formula/astroid.rb [hunk 1]
```diff
@@ -19,25 +20,18 @@ class Astroid < Formula
...
-      "-j#{ENV.make_jobs}",
+      "-Ddisable-embedded-editor=true",
+      "-Ddisable-plugins=true",
+      "-Ddisable-tests=true",
     ]
-    # overwrite --release if --HEAD with `git` magic variable for SCons
-    args += [ "--release=git" ] if build.head?
-    scons "install", *args
+    system "meson", "build", *args
+    system "sed", "-i", "-e", "s/boost_log\ /boost_log-mt\ /g", "build/build.ninja"
```
Comment (gauteh) [inline Formula/astroid.rb:33]: Only thing I can think of; will this work or fail or, worst case, do something wrong once https://github.com/mesonbuild/meson/issues/2436 gets fixed?

**Review by gauteh — COMMENTED** [review]
Looks good to me! But I don't have a Mac to test it on atm.

File: Formula/astroid.rb [hunk 2]
```diff
@@ -8,7 +8,8 @@ class Astroid < Formula
   sha256 "7d58a813a9e8f840475226a254743e0caf50f1baf830256ce17e135b71f34714"
   head "https://github.com/astroidmail/astroid.git"
 
-  depends_on "scons" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
   depends_on "libsass"
   depends_on "libpeas"
```
Comment (gauteh) [inline Formula/astroid.rb:14]: Don't think you need this if you disable plugins. Are you sure GIR generation still breaks?

File: Formula/astroid.rb (hunk 1 above)
Comment (yeled) [reply Formula/astroid.rb:33]: Yes this will probably break. Considering this is HEAD I think it's ok.

File: Formula/astroid.rb (hunk 2 above)
Comment (yeled) [reply Formula/astroid.rb:14]: Sadly, yes.

```
[51/65] Generating Astroid-0.1.gir with a custom command.
FAILED: Astroid-0.1.gir
/usr/local/opt/gobject-introspection/bin/g-ir-scanner ../src/plugin/astroid_activatable.c ../src/plugin/thread_view_activatable.c ../src/plugin/thread_index_activatable.c ../src/plugin/astroid_activatable.h ../src/plugin/thread_view_activatable.h ../src/plugin/thread_index_activatable.h ../src/plugin/gir_main.c -D_REENTRANT -I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/pcre/8.41/include -I/usr/local/Cellar/glib/2.54.1/include/glib-2.0 -I/usr/local/Cellar/glib/2.54.1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/gobject-introspection/1.54.1/include/gobject-introspection-1.0 --no-libtool --namespace=Astroid --nsversion=0.1 --warn-all --output Astroid-0.1.gir -I/private/tmp/astroid-20171007-75146-18ogk1r/ -I/private/tmp/astroid-20171007-75146-18ogk1r/build/ -I./src -I../src --include=GObject-2.0 --include=GMime-3.0 --cflags-begin -D_REENTRANT -I/usr/local/Cellar/pcre/8.41/include -I/usr/local/Cellar/glib/2.54.1/include/glib-2.0 -I/usr/local/Cellar/glib/2.54.1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/gmime/3.0.2/include/gmime-3.0 -I/usr/local/Cellar/glib/2.54.1/include/gio-unix-2.0/ -I/usr/local/Cellar/libpng/1.6.32/include/libpng16 -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/Cellar/fontconfig/2.12.6/include -I/usr/local/Cellar/pixman/0.34.0_1/include/pixman-1 -I/usr/local/Cellar/cairo/1.14.10/include/cairo -I/usr/local/Cellar/gdk-pixbuf/2.36.11/include/gdk-pixbuf-2.0 -I/usr/local/Cellar/atk/2.26.0/include/atk-1.0 -I/usr/local/Cellar/libepoxy/1.4.3/include -I/usr/local/Cellar/graphite2/1.3.10/include -I/usr/local/Cellar/harfbuzz/1.5.1/include/harfbuzz -I/usr/local/Cellar/pango/1.40.12/include/pango-1.0 -I/usr/local/Cellar/gtk+3/3.22.24/include/gtk-3.0 -I/usr/local/Cellar/zlib/1.2.11/include -I/usr/local/Cellar/gtk+3/3.22.24/include/gtk-3.0/unix-print -I/usr/local/Cellar/libsigc++/2.10.0/include/sigc++-2.0 -I/usr/local/Cellar/libsigc++/2.10.0/lib/sigc++-2.0/include -I/usr/local/Cellar/glibmm/2.54.1/include/giomm-2.4 -I/usr/local/Cellar/glibmm/2.54.1/lib/giomm-2.4/include -I/usr/local/Cellar/glibmm/2.54.1/include/glibmm-2.4 -I/usr/local/Cellar/glibmm/2.54.1/lib/glibmm-2.4/include -I/usr/local/Cellar/cairomm/1.12.2/include/cairomm-1.0 -I/usr/local/Cellar/cairomm/1.12.2/lib/cairomm-1.0/include -I/usr/local/Cellar/pangomm/2.40.1/include/pangomm-1.4 -I/usr/local/Cellar/pangomm/2.40.1/lib/pangomm-1.4/include -I/usr/local/Cellar/atkmm/2.24.2/include/atkmm-1.6 -I/usr/local/Cellar/gtkmm3/3.22.2/include/gtkmm-3.0 -I/usr/local/Cellar/gtkmm3/3.22.2/lib/gtkmm-3.0/include -I/usr/local/Cellar/gtkmm3/3.22.2/include/gdkmm-3.0 -I/usr/local/Cellar/gtkmm3/3.22.2/lib/gdkmm-3.0/include -I/usr/local/Cellar/libsoup/2.60.0/include/libsoup-2.4 -I/usr/local/Cellar/webkitgtk@2.4.11/2.4.11/include/webkitgtk-3.0 -I/usr/include/libxml2 -I/usr/local/Cellar/libtasn1/4.12/include -I/usr/local/Cellar/nettle/3.3/include -I/usr/local/Cellar/p11-kit/0.23.9/include/p11-kit-1 -I/usr/local/Cellar/gnutls/3.5.15/include -I/usr/local/Cellar/pcre2/10.30/include -I/usr/local/Cellar/vte3/0.50.1/include/vte-2.91 -I/usr/local/Cellar/libsass/3.4.5/include -I/usr/local/Cellar/libffi/3.2.1/lib/libffi-3.2.1/include -I/usr/local/Cellar/gobject-introspection/1.54.1/include/gobject-introspection-1.0 -I/usr/local/Cellar/libpeas/1.22.0/include/libpeas-1.0 --cflags-end -L/usr/local/Cellar/glib/2.54.1/lib -L/usr/local/opt/gettext/lib -L/usr/local/Cellar/gmime/3.0.2/lib --extra-library=gmime-3.0 --extra-library=gio-2.0 --extra-library=gobject-2.0 --extra-library=glib-2.0 --extra-library=intl -L/usr/local/lib --extra-library=boost_thread-mt --extra-library=boost_filesystem --extra-library=boost_program_options --extra-library=boost_log_setup --extra-library=boost_log-mt --extra-library=boost_date_time --extra-library=boost_system -L/usr/local/Cellar/cairo/1.14.10/lib -L/usr/local/Cellar/gdk-pixbuf/2.36.11/lib -L/usr/local/Cellar/atk/2.26.0/lib -L/usr/local/Cellar/pango/1.40.12/lib -L/usr/local/Cellar/gtk+3/3.22.24/lib --extra-library=gtk-3 --extra-library=gdk-3 --extra-library=pangocairo-1.0 --extra-library=pango-1.0 --extra-library=atk-1.0 --extra-library=cairo-gobject --extra-library=cairo --extra-library=gdk_pixbuf-2.0 -L/usr/local/Cellar/libsigc++/2.10.0/lib -L/usr/local/Cellar/glibmm/2.54.1/lib -L/usr/local/Cellar/cairomm/1.12.2/lib -L/usr/local/Cellar/pangomm/2.40.1/lib -L/usr/local/Cellar/atkmm/2.24.2/lib -L/usr/local/Cellar/gtkmm3/3.22.2/lib --extra-library=gtkmm-3.0 --extra-library=atkmm-1.6 --extra-library=gdkmm-3.0 --extra-library=giomm-2.4 --extra-library=pangomm-1.4 --extra-library=glibmm-2.4 --extra-library=cairomm-1.0 --extra-library=sigc-2.0 -L/usr/local/Cellar/libsoup/2.60.0/lib -L/usr/local/Cellar/webkitgtk@2.4.11/2.4.11/lib --extra-library=webkitgtk-3.0 --extra-library=soup-2.4 --extra-library=javascriptcoregtk-3.0 --extra-library=notmuch -L/usr/local/Cellar/zlib/1.2.11/lib -L/usr/local/Cellar/gnutls/3.5.15/lib -L/usr/local/Cellar/pcre2/10.30/lib -L/usr/local/Cellar/vte3/0.50.1/lib --extra-library=vte-2.91 --extra-library=z --extra-library=pcre2-8 --extra-library=gnutls -L/usr/local/Cellar/libsass/3.4.5/lib --extra-library=sass -L/usr/local/Cellar/gobject-introspection/1.54.1/lib -L/usr/local/Cellar/libpeas/1.22.0/lib --extra-library=peas-1.0 --extra-library=gmodule-2.0 --extra-library=girepository-1.0 -I./src -I../src --add-include-path=./src --add-include-path=../src --program /private/tmp/astroid-20171007-75146-18ogk1r/build/gir_main
Traceback (most recent call last):
  File "/usr/local/opt/gobject-introspection/bin/g-ir-scanner", line 66, in <module>
    sys.exit(scanner_main(sys.argv))
  File "/usr/local/Cellar/gobject-introspection/1.54.1/lib/gobject-introspection/giscanner/scannermain.py", line 538, in scanner_main
    ss = create_source_scanner(options, args)
  File "/usr/local/Cellar/gobject-introspection/1.54.1/lib/gobject-introspection/giscanner/scannermain.py", line 446, in create_source_scanner
    ss.parse_files(filenames)
  File "/usr/local/Cellar/gobject-introspection/1.54.1/lib/gobject-introspection/giscanner/sourcescanner.py", line 262, in parse_files
    self._parse(headers)
  File "/usr/local/Cellar/gobject-introspection/1.54.1/lib/gobject-introspection/giscanner/sourcescanner.py", line 307, in _parse
    self._cpp_options)
  File "/usr/local/Cellar/gobject-introspection/1.54.1/lib/gobject-introspection/giscanner/ccompiler.py", line 210, in preprocess
    extra_postargs=extra_postargs)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/unixccompiler.py", line 114, in preprocess
    raise CompileError, msg
distutils.errors.CompileError: command 'clang' terminated by signal 10
```

**Comment by gauteh** [comment]
Ok, this might be a meson bug (not that it worked on SCons).

**Pull request merged**